open_instruct/grpo_fast.py

```diff
@@ -1823,11 +1823,17 @@ def reward_fn(
                     datasets,
                     reward_mult=args.verification_reward,
                 )
+                if len(verifiable_rewards) != len(scores):
+                    raise ValueError(f"{len(verifiable_rewards)=} != {len(scores)=}")
+                # slightly complex combo of good outputs and additive format reward
                 for i in range(len(verifiable_rewards)):
-                    if good_outputs[i] and args.only_reward_good_outputs:
-                        scores[i] = verifiable_rewards[i]
-                    elif not args.only_reward_good_outputs:
-                        scores[i] = verifiable_rewards[i] + scores[i]
+                    if not args.only_reward_good_outputs or (good_outputs[i] and args.only_reward_good_outputs):
```

Comment:
`only_reward_good_outputs` would be nice to name something related to tools. Took me a bit to grok skimming the code. Maybe not worth the effort.